File: solutions/2786-find-the-longest-semi-repetitive-substring/find-the-longest-semi-repetitive-substring.py
class Solution:
    def longestSemiRepetitiveSubstring(self, s: str) -> int:
        if len(s) == 1:
            return 1
        
        l, r, ans, cnt = 0, 1, 0, 0
        
        while r < len(s):
            cnt += (s[r] == s[r-1])
            
            while l <= r and cnt > 1:
                cnt -= (s[l] == s[l+1])
                l += 1
                
            ans = max(ans, r - l + 1)
            r += 1
            
        return ans

File: solutions/2786-find-the-longest-semi-repetitive-substring/find-the-longest-semi-repetitive-substring.cpp
class Solution {
public:
    int longestSemiRepetitiveSubstring(string s) {
        if(s.size()==1) return 1;
        int l=0,r=1, ans=0,cnt=0;
        while(r<s.size()){
            cnt+=(s[r]==s[r-1]);
            while(l<=r && cnt>1){
                cnt-=(s[l]==s[l+1]);
                l++;
            }
            ans=max(ans, r-l+1);
            r++;
        }
        return ans;
    }
};